core/src/main/java/org/jruby/ir/representations/BasicBlock.java

```diff
@@ -115,24 +119,43 @@ public boolean isEmpty() {
         return instrs.isEmpty();
     }
 
-    // FIXME: inline branch fixes this by using callsiteID and not ipc.  Temporary change until it is
-    // merged (this is only for inlining and inlining does not work on master currently).
-    public BasicBlock splitAtInstruction(Instr splitPoint, Label newLabel, boolean includeSplitPointInstr) {
+    /**
+     * What site object contains this callsiteId or die trying.
+     * @param callsiteId to be found
+     * @return the Site instance (CallBase or YieldInstr)
+     */
+    public Site siteOf(long callsiteId) {
+        for (Instr instr: instrs) {
+            if (instr instanceof Site && ((Site) instr).getCallSiteId() == callsiteId) return (Site) instr;
```

Comment:
hmm, seems (above) might matter - better of using an AtomicLong instead of long++